frontend/javascripts/viewer/view/action-bar/starting_job_modals.tsx

```diff
@@ -712,28 +726,64 @@ function useCurrentlySelectedBoundingBox(
   return currentlySelectedBoundingBox;
 }
 
+// This function mirrors the selection of the mag
+// in voxelytics/worker/job_utils/voxelytics_utils.py select_mag_for_model_prediction
+const getBestFittingMagComparedToTrainingDS = (
+  colorLayer: APIDataLayer,
+  datasetScaleMag1: VoxelSize,
+  jobType: APIJobType.INFER_MITOCHONDRIA | APIJobType.INFER_NEURONS | APIJobType.INFER_NUCLEI,
+) => {
+  if (jobType === APIJobType.INFER_MITOCHONDRIA) {
+    // infer_mitochondria_model always infers on the finest mag of the current dataset
+    const magInfo = getMagInfo(colorLayer.resolutions);
+    return magInfo.getFinestMag();
+  }
+  const modelScale = MEAN_VX_SIZE[jobType];
+  let closestMagOfCurrentDS = colorLayer.resolutions[0];
+  let bestDifference = [
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+  ];
+
+  const datasetScaleInNm = convertVoxelSizeToUnit(datasetScaleMag1);
```

Comment:
I agree that default arguments can be very helpful, but in this case it makes it harder to understand the code, I'd say. The function name doesn't indicate which unit the voxel size is converted to so one needs to look at the function definition to make sense of it. I'd suggest to remove the "nm" parameter default and explicitly include it in the function calls (same for the tests).